lake/api/remote.go

```diff
@@ -65,29 +77,23 @@ func (r *RemoteSession) CreatePool(ctx context.Context, name string, layout orde
 	}
 	var meta lake.BranchMeta
 	err = unmarshal(res, &meta)
-	return meta.PoolConfig.ID, err
+	return meta.Pool.ID, err
 }
 
-func (r *RemoteSession) CreateBranch(ctx context.Context, poolID ksuid.KSUID, name string, parent, at ksuid.KSUID) (ksuid.KSUID, error) {
-	res, err := r.conn.BranchPost(ctx, poolID, api.BranchPostRequest{
-		Name:     name,
-		ParentID: parent.String(),
-		At:       at.String(),
+func (r *RemoteSession) CreateBranch(ctx context.Context, poolID ksuid.KSUID, name string, at ksuid.KSUID) error {
+	_, err := r.conn.BranchPost(ctx, poolID, api.BranchPostRequest{
+		Name:   name,
+		Commit: at.String(),
 	})
```

Comment:
```suggestion
	r, err := r.conn.BranchPost(ctx, poolID, api.BranchPostRequest{
		Name:   name,
		Commit: at.String(),
	})
	if err == nil {
		r.Body.Close()
	}
```
I started this pattern in a lot of places, but we need to still close the response if the err is nil otherwise the connection will not close. See https://pkg.go.dev/net/http#Client.Do

We should probably change the connection library to allows return a response Body so we can go without the error check.

Reply:
Made the change but I had the opposite reaction on the client lib ... the client only needs the response body if it's going to stream back the result.  Otherwise, the library should do all the boiler plate to unmarshal expected structure and return meaningful values instead of a generic response object.

Reply:
I agree.